### ISSUE TITLE
Remove dependency of `FunctionSpaceElement` on `Operator`

### ISSUE THREAD

**Issue by kohr-h** [odlgroup/odl#949]
This was a lazy way of implementing function spaces in the first space. Probably the usefulness of all the `Operator` stuff is marginal. Removing this dependency would greatly improve the flexibility of vectorized functions, e.g.
```python
>>> def f(x, c):
...     return sum(x) + c
...
>>> fspace = odl.FunctionSpace(odl.IntervalProd([0, 0], [1, 1]))
>>> f_elem = fspace.element(f)
>>> points = np.array([[0, 0], [0.5, 0], [1, 1]]).T
>>> f_elem(points, c=0)
array([ 0. ,  0.5,  2. ])
>>> f_elem(points, c=2)
array([ 2. ,  2.5,  4. ])
```
This is not possible currently because `Operator` has a very harsh signature restriction that is not needed here.

**Comment by adler-j** [comment]
Fine by me!

**Comment by kohr-h** [comment]
Closed by #1088
